### PR TITLE
Add libcantera-devel documentation

### LINKED ISSUE #180: Document installation/use of libcantera-devel conda package
We need some documentation of:
- What this package provides
- What platforms are supported / current limitations
- How to install it
- How to use it when compiling / linking C++ (or Fortran?) applications

### PR DESCRIPTION
**Pull request by ischoegl**
Add documentation for `libcantera-devel`.

Fixes #180

While Cantera/conda-recipes#35 fixes some minor paper cuts, the instructions themselves work.

### CODE REVIEW

File: pages/install/conda-install.rst [hunk 1]
```diff
@@ -206,3 +206,80 @@ by running the commands:
...
+
+Development (C++ & Fortran 90) Interface
+========================================
+
+**Warning:** The ``conda`` package for Cantera's development interface is experimental.
+For the most recent version of the package, use the ``cantera/label/dev`` channel
+instead of the stable ``cantera`` channel.
+
+Cantera's development interface can be installed from the ``cantera`` channel. In this
+example, the command will create a new conda environment named ``ct-env``. From the
```
Comment (bryanwweber) [inline pages/install/conda-install.rst:218]:
```suggestion
example, the command will create a new conda environment named ``ct-dev``. From the
```

File: pages/install/conda-install.rst [hunk 2]
```diff
@@ -206,3 +206,80 @@ by running the commands:
...
+command line (or the Anaconda Prompt on Windows), run:
+
+.. code:: shell
+
+   conda create --name ct-dev --channel cantera libcantera-devel
```
Comment (bryanwweber) [inline pages/install/conda-install.rst:223]:
```suggestion
   conda create --name ct-dev --channel cantera/label/dev libcantera-devel
```

At least until we feel comfortable removing the warning at the top of the section?

**Review by bryanwweber — CHANGES_REQUESTED** [review]
Thanks! One small correct, and a question.

File: pages/install/conda-install.rst (hunk 1 above)
Comment (ischoegl) [reply pages/install/conda-install.rst]: copy-pasta

File: pages/install/conda-install.rst (hunk 2 above)
Comment (ischoegl) [reply pages/install/conda-install.rst]: Also added the description below.

**Comment by ischoegl** [comment]
@bryanwweber ... thanks for the comments! Things should be taken care of.

File: pages/install/conda-install.rst [hunk 3]
```diff
@@ -206,3 +206,80 @@ by running the commands:
...
+command line (or the Anaconda Prompt on Windows), run:
+
+.. code:: shell
+
+   conda create --name ct-dev --channel cantera/label/dev libcantera-devel
+
+This will create an environment with Cantera's development interface. In this case,
+the addition of ``--channel cantera/label/dev`` ensures that the package is pulled
+from the most recent available Cantera version. Note that ``label/dev`` refers to the
+experimental development *channel* of Cantera, and not the development *interface*.
```
Comment (bryanwweber) [inline pages/install/conda-install.rst:228]: Nice catch!

**Review by bryanwweber — APPROVED** [review]
Thanks @ischoegl!

**Pull request merged**